src/cluster.py

```diff
@@ -48,34 +44,54 @@ class SwitchoverFailedError(Exception):
 class Patroni:
     """This class handles the bootstrap of a PostgreSQL database through Patroni."""
 
-    def __init__(self, unit_ip: str):
-        self.unit_ip = unit_ip
+    pass
 
-    def bootstrap_cluster(
+    def __init__(
         self,
+        unit_ip: str,
+        storage_path: str,
         cluster_name: str,
         member_name: str,
+        planned_units: int,
+        peers_ips: List[str],
         superuser_password: str,
         replication_password: str,
-    ) -> bool:
-        """Bootstrap a PostgreSQL cluster using Patroni.
+    ):
+        """Initialize the Patroni class.
 
         Args:
+            unit_ip: IP address of the current unit
+            storage_path: path to the storage mounted on this unit
             cluster_name: name of the cluster
             member_name: name of the member inside the cluster
+            peers_ips: IP addresses of the peer units
+            planned_units: number of units planned for the cluster
             superuser_password: password for the postgres user
             replication_password: password for the user used in the replication
         """
+        self.unit_ip = unit_ip
+        self.storage_path = storage_path
+        self.cluster_name = cluster_name
+        self.member_name = member_name
+        self.planned_units = planned_units
+        self.peers_ips = peers_ips
+        self.superuser_password = superuser_password
+        self.replication_password = replication_password
+
+    def bootstrap_cluster(self, replica: bool = False) -> bool:
+        """Bootstrap a PostgreSQL cluster using Patroni."""
         # Render the configuration files and start the cluster.
-        self._change_owner(STORAGE_PATH)
-        self._render_patroni_yml_file(
-            cluster_name, member_name, superuser_password, replication_password
-        )
+        self.configure_patroni_on_unit(replica)
+        return self.start_patroni()
+
+    def configure_patroni_on_unit(self, replica: bool = False):
+        """Configure Patroni (configuration files and service) on the unit."""
```

Comment:
nit: add arguments to docstring (not sure the effects of `replica` argument)

Reply:
Added in https://github.com/canonical/postgresql-operator/pull/11/commits/d2910b257fdfe713c0aa918ef5c470f439751997.